Disable autocompletion for login forms and password fields

mfrister opened the pull request:
OpenProject Issue: https://www.openproject.org/issues/783

mfrister committed:
Disable autocompletion for login forms and password fields

mfrister committed:
Disable autocomplete on forms with login password fields

Includes login, register, change password and user admin pages.

mfrister commented:
I just added a commit disabling autocomplete for all forms containing login passwords, I could find by searching for password_field.
This does not include e.g. LDAP or Subversion configuration.

Seems all major browsers support autocomplete="off" on form tags since a while:
https://developer.mozilla.org/en-US/docs/Mozilla/How_to_Turn_Off_Form_Autocompletion
http://stackoverflow.com/questions/3868299/is-autocomplete-off-compatible-with-all-modern-browsers

ulferts commented:
I believe it would be better for some of the forms to not disable autocompletion all together. Because doing it will also disable autocompletion on inputs for which autocompletion is ok.

As an example:
I believe autocompletion should be disabled on the account/register view for the password and password_confirmation field. One could argue about the login field. For all the other fields autocompletion is ok, as far as I am concerned. Why should we prevent autocompletion on "firstname", "lastname", "email", ...? 

Building on the example, I propose to disable autocompletion not on form level but for individual inputs on the following views:
- account/register
- users/new
- users/edit

mfrister committed:
Reenable autocompletion for register form except password

Also remove superfluous autocompletion disabling of the
user/edit and user/new password fields.

mfrister commented:
As discussed, reenabled autocompletion for register (except password fields) and removed superfluous attributes for the user edit/new pages.
